cmds/core/ping/ping_test.go

```diff
@@ -0,0 +1,232 @@
+// Copyright 2021 the u-root Authors. All rights reserved
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package main
+
+import (
+	"fmt"
+	"net"
+	"testing"
+	"time"
+)
+
+type MyConn struct {
```

Comment:
You can simplify this a  bit if you do this
type myConn struct {
net.IPConn
testRun string
}

Then, the ONLY functions you need implement are Read, and Write. This makes life simpler, should the Conn interface ever change, you are insulated from that change by the embedded net.Conn

Since testRun probably should not be exported, and in fact MyConn should not be exported, they might as well be lower case.

Reply:
Thank you for that!